SFR 461 edition date range

mwbenowitz opened the pull request:
This generates a string for each work result in a search response stored in the `edition_range` field which represents the full date range for the editions present within the work. If edition publication dates could not be found it will return '????' for a date value.

This is implemented as a custom filter function and assumes that all publication dates should be represented as GMT, as they have no times associated with them.

Additional tests have been added to cover the new methods.

mwbenowitz committed:
SFR-461 Add Edition range for work results

This generates a string for each work result in a search response stored in the `edition_range` field which represents the full date range for the editions present within the work. If edition publication dates could not be found it will return '????' for a date value.

This is implemented as a custom filter function and assumes that all publication dates should be represented as GMT, as they have no times associated with them.

gkallenberg reviewed (commented):
Looks good, but the tests return 1 failure. Am getting an error when attempting to run tests against local Elasticsearch (error about localhost:9200 not being available). I don't think that's stipulated in the docs. Does ES need to be running for the test environment?

mwbenowitz committed:
SFR-461 Update tests with Nock

Forgot to mock ElasticSearch responses for routing testing. This
utilizes `nock` to ensure that the tests are mocked and adds a line
to ensure the `env` variables are properly loaded for mocking.

mwbenowitz commented:
@gkallenberg Updated the tests, they should pass now. It was a combination of two things:

- First, I had forgotten to commit some code mocking the ElasticSearch response
- Second, the `env` variables weren't being properly set, which why it was trying to hit `localhost`
